src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -213,7 +214,7 @@ public Task<RestoreResult> ExecuteAsync()
             return new KeyValuePair<CacheFile,bool>(cacheFile, _noOp) ;
         }
 
-        private bool VerifyAssetsAndMSBuildFilesArePresent(RestoreRequest request)
+        private bool VerifyAssetsAndMSBuildFilesAndPackagesArePresent(RestoreRequest request)
```

Comment:
Move this and all helpers to a static utility class. You'll be able to unit test it easier and it will simplify the RestoreCommand class (which is already massive).